src/Build/Resources/MSBuildAssemblyFileVersion.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Build.Internal
+{
+    /// <summary>
+    /// Gets the <see cref="AssemblyFileVersionAttribute"/> of Microsoft.Build.dll.
+    /// </summary>
+    internal sealed class MSBuildAssemblyFileVersion
+    {
+        private static readonly Lazy<MSBuildAssemblyFileVersion> MSBuildAssemblyFileVersionLazy = new Lazy<MSBuildAssemblyFileVersion>(GetMSBuildAssemblyFileVersion, isThreadSafe: true);
+
+        private MSBuildAssemblyFileVersion(string majorMinor)
+        {
+            MajorMinor = majorMinor;;
+        }
+
+        /// <summary>
+        /// Gets a singleton instance of <see cref="MSBuildAssemblyFileVersion"/>.
+        /// </summary>
+        public static MSBuildAssemblyFileVersion Instance
+        {
+            get { return MSBuildAssemblyFileVersionLazy.Value; }
+        }
+
+        /// <summary>
+        /// Gets the assembly file version in the format major.minor.
+        /// </summary>
+        public string MajorMinor { get; set; }
+
+        private static MSBuildAssemblyFileVersion GetMSBuildAssemblyFileVersion()
+        {
+            string versionString = typeof(MSBuildAssemblyFileVersion).GetTypeInfo()?.Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version;
+
+            Version version;
+
+            if (String.IsNullOrEmpty(versionString) || !Version.TryParse(versionString, out version))
+            {
+                // Fall back to the constant AssemblyVersion
+                return new MSBuildAssemblyFileVersion(Constants.AssemblyVersion);
+            }
+
+            return new MSBuildAssemblyFileVersion($"{version.Major}.{version.Minor}");
```

Comment:
This seems reasonable but we might want to go three-part here, so that a build extension author can say "you n need to install 15.23.1 because it has an engine fix we depend on". We probably don't want to go all the way to 4-part, since the NB.GV-generated 4th part is based on the commit hash and not comparable.

Reply:
At the moment, we're not using NBGV for .NET Core so we only set it as `15.6`.  After we've moved to the new SDK, we'll be using NBGV for .NET Core and I could change this to 3 part.  But would it be better to only be 2 part for this:

```xml
<Error
    Condition=" '$(MSBuildVersion)' &lt; '15.7' "
    Text="You must build this code base with MSBuild 15.7 or above"
 />
```

Since we don't have a separate property for major, minor, and build, string comparison is the only choice.  I'm worried a 3 part version would make this harder, what do you think?

Reply:
Won't that go into version-comparison mode, and be able to handle up to four-part versions?

Reply:
Rainer is right, we `Version.TryParse()` all strings in conditions so there is no harm in the version being major.minor.build.

At the moment. .NET core versions will `0` for the build number until we move to the new SDK.